dspace-api/src/main/java/org/dspace/service/impl/HttpConnectionPoolService.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.service.impl;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.http.HeaderElement;
+import org.apache.http.HeaderElementIterator;
+import org.apache.http.HttpResponse;
+import org.apache.http.conn.ConnectionKeepAliveStrategy;
+import org.apache.http.conn.HttpClientConnectionManager;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.message.BasicHeaderElementIterator;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.protocol.HttpContext;
+import org.dspace.services.ConfigurationService;
+
+/**
+ * Factory for HTTP clients sharing a pool of connections.
+ *
+ * @author Mark H. Wood <mwood@iupui.edu>
+ */
+@Named
+@Singleton
+public class HttpConnectionPoolService {
```

Comment:
Since the only Solr-specific thing is the names of the configuration properties, I've made the leading part of those names a constructor argument.  We can now have a pool for Solr configured by `solr.client.*`, a pool for X configured by `X.client` and so forth.  Code that wants a pool can have the one it needs injected by `id`.